lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php

```diff
@@ -77,14 +81,29 @@ public function setReferenceRepository(ReferenceRepository $referenceRepository)
     public function execute(array $fixtures, $append = false)
     {
         $executor = $this;
-        $this->em->wrapInTransaction(static function (EntityManagerInterface $em) use ($executor, $fixtures, $append) {
+        if ($this->singleTransaction) {
+            $this->em->wrapInTransaction(static function (EntityManagerInterface $em) use ($executor, $fixtures, $append) {
+                if ($append === false) {
+                    $executor->purge();
+                }
+
+                foreach ($fixtures as $fixture) {
+                    $executor->load($em, $fixture);
+                }
+            });
+        } else {
             if ($append === false) {
-                $executor->purge();
+                $this->em->wrapInTransaction(static function () use ($executor) {
+                    $executor->purge();
+                });
             }
 
+            $em = $this->em;
             foreach ($fixtures as $fixture) {
-                $executor->load($em, $fixture);
+                $this->em->wrapInTransaction(static function () use ($executor, $fixture, $em) {
+                    $executor->load($em, $fixture);
+                });
```

Comment:
```suggestion
                $this->em->wrapInTransaction(static function (EntityManagerInterface $em) use ($executor, $fixture) {
                    $executor->load($em, $fixture);
                });
```

`wrapInTransaction()` will pass the EM to your closure.

Reply:
Good to know, thank you